### PR TITLE
Rename Measure and Measurement subclasses.

### PR DESCRIPTION
**Pull request by songy23**
Rename to follow the name convention on Tag/TagKey:
- DoubleMeasure -> MeasureDouble
- LongMeasure -> MeasureLong
- DoubleMeasurement -> MeasurementDouble
- LongMeasure -> MeasurementLong 

This PR is built on top of https://github.com/census-instrumentation/opencensus-java/pull/405, I'll rebase this PR once I've merged PR https://github.com/census-instrumentation/opencensus-java/pull/405.

### CODE REVIEW

File: core/src/main/java/io/opencensus/stats/Measurement.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+/**
+ * Immutable representation of a Measure.
+ */
+@Immutable
+public abstract class Measurement {
+  /**
+   * Applies the given match function to the underlying data type.
+   */
+  public abstract <T> T match(Function<MeasurementDouble, T> p0, Function<MeasurementLong, T> p1);
```
Comment (dinooliva) [inline core/src/main/java/io/opencensus/stats/Measurement.java]: You should use '? super' here

File: core_impl/src/main/java/io/opencensus/stats/MeasureToViewMap.java [hunk 2]
```diff
@@ -90,18 +93,30 @@ synchronized void registerView(ViewDescriptor viewDescriptor, Clock clock) {
...
-  synchronized void record(StatsContextImpl tags, MeasurementMap stats) {
-    for (MeasurementValue mv : stats) {
+  synchronized void record(StatsContextImpl tags, MeasureMap stats) {
+    Iterator<Measurement> iterator = stats.iterator();
+    while (iterator.hasNext()) {
+      Measurement measurement = iterator.next();
       Collection<MutableView> views =
-          mutableMap.get(mv.getMeasurement().getName());
+          mutableMap.get(measurement.getMeasure().getName());
+      Object value = measurement.match(new GetDoubleValueFunc(), new GetLongValueFunc());
```
Comment (dinooliva) [inline core_impl/src/main/java/io/opencensus/stats/MeasureToViewMap.java]: Consider making the a Double and having GetLongValueFunc() throw and exception.

Also, you don't really need a new instance of GetDoubleValueFunc() and GetLongValueFunc() for each use.

File: core/src/main/java/io/opencensus/stats/Measurement.java (hunk 1 above)
Comment (songy23) [inline core/src/main/java/io/opencensus/stats/Measurement.java]: Oh sorry but this PR is built on top of https://github.com/census-instrumentation/opencensus-java/pull/405, so these parts will be updated in https://github.com/census-instrumentation/opencensus-java/pull/405. I'll rebase this PR once I've merged https://github.com/census-instrumentation/opencensus-java/pull/405, and for now please review  the last commit only.

File: core_impl/src/main/java/io/opencensus/stats/MeasureToViewMap.java (hunk 2 above)
Comment (songy23) [inline core_impl/src/main/java/io/opencensus/stats/MeasureToViewMap.java]: Same as above, I've updated this in PR https://github.com/census-instrumentation/opencensus-java/pull/405/files#diff-a5ab43befefb8a0d5c22633f8af615ceR132. Sorry for the confusion.

**Comment by songy23** [comment]
Rebased, merging soon...

**Pull request merged**